dev-tools/generate_notice.py

```diff
@@ -10,7 +11,7 @@ def read_file(filename):
         print("File not found {}".format(filename))
         return ""
 
-    with open(filename, 'r') as f:
+    with open(filename, 'rb') as f:
```

Comment:
NB: one of the license files generated this error on read using Python 3:

```
UnicodeDecodeError: 'utf-8' codec can't decode byte 0x93 at position ...
```

Switching file operations to binary mode fixes this issue, but all strings must be bytes, necessitating the `six.b` additions.